### PR TITLE
Spec frontend tests

### PR DESCRIPTION
**Pull request by Dampolo03**
## Done
- Added test cases for the client-side.

In the root directory, run
- `npm run test-js` to run all test suites.
- `npm run test-coverage-js` to get tests coverage.

### CODE REVIEW

**Comment by andesol** [comment]
Waiting for this one to be unblocked #90, so this branch can be updated

File: client/tests/Filters.test.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import crypto from "crypto";
+import Filters from "../Filters";
+import { testSpec, testTeams } from "./__mocks__/utils";
+import { fireEvent, render, screen } from "@testing-library/react";
+
+Object.defineProperty(global, "crypto", {
```
Comment (andesol) [inline client/tests/Filters.test.tsx:6]: Just out of curiosity, what is this for?

**Review by andesol — COMMENTED** [review]
Great! LGTM

**Review by andesol — APPROVED** [review]

File: client/tests/Filters.test.tsx (hunk 1 above)
Comment (Dampolo03) [reply client/tests/Filters.test.tsx:6]: @andesol Had some funny error that looked like tthis - `Error: Uncaught [ReferenceError: crypto is not defined]` coming from the `qs` library in the project, if I can remember correctly.

Comment (Dampolo03) [reply client/tests/Filters.test.tsx:6]: @andesol Had some funny error that looked like tthis - `Error: Uncaught [ReferenceError: crypto is not defined]` coming from the `window.crypto` API, if I can remember correctly.

**Pull request merged**